src/Tools/dotnet-monitor/Extensibility/FileProvider/FolderFileProvider.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Primitives;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Extensibility
+{
+    /// <summary>
+    /// This class provides an implementation of <see cref="IFileProvider"/> with a root directory that does not need to exist at construction time.
+    /// </summary>
+    public class FolderFileProvider : IFileProvider
+    {
+        private IFileProvider _fileProvider = null;
+        private readonly string _rootPath;
+
+        private IFileProvider BaseProvider
+        {
+            get
+            {
+                if (_fileProvider == null)
+                {
+                    if (Directory.Exists(_rootPath))
+                    {
+                        _fileProvider = new PhysicalFileProvider(_rootPath);
```

Comment:
Any unusual behavior on a mounted drive?

Reply:
Not that I'm aware of right now, but I'll test this further when I add tests.